Framework/include/QualityControl/TaskInterface.h

```diff
@@ -74,17 +84,22 @@ class TaskInterface
   // Setters and getters
   void setObjectsManager(std::shared_ptr<ObjectsManager> objectsManager);
   void setName(const std::string& name);
+  void setCcdbHost(const std::string& url);
   void setCustomParameters(const std::unordered_map<std::string, std::string>& parameters);
   const std::string& getName() const;
 
  protected:
   std::shared_ptr<ObjectsManager> getObjectsManager();
+  TObject* retrieveCondition(std::string path, std::map<std::string, std::string> metadata = {}, long timestamp = -1);
+
   std::unordered_map<std::string, std::string> mCustomParameters;
 
  private:
   // TODO should we rather have a global/singleton for the objectsManager ?
   std::shared_ptr<ObjectsManager> mObjectsManager;
   std::string mName;
+  std::string mCcdbUrl = "";
+  std::shared_ptr<o2::ccdb::CcdbApi> mCcdbApi;
```

Comment:
Here we should probably keep only `mCcdbApi` to avoid confusion. S

Reply:
This is needed to initialize mCcdbApi in a initializaiton function, instead of constructor.

Reply:
I got rid of it by having `TaskInterface::loadCcdb()` take the ccdb url as an argument